### DISCUSSION TITLE
The ERROR of no matching GDS/OAS cell

### DISCUSSION THREAD

**Discussion by 706697916** [The-OpenROAD-Project/OpenROAD-flow-scripts#1288] [General]
I changed the contents of PDK file . The 6_1_merged.gds and 6_final.def can be generated.  But the ERROR occured when the Makefile executes the code on line 731.  I have added the tech.lef in the lyt file and it includes the via cell. I think this problem should not happen.
Below is the ERROR log and the corresponding Makefile code.
<img width="948" alt="bf6ddb725ca7a879b8d1f5c9a89d30d" src="https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/assets/70256434/a82a97c3-46a3-4d2e-8a4d-660d800165e4">
<img width="996" alt="7f386ec830ddf077f00cf78f46bb0c2" src="https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/assets/70256434/b6c47b60-ebab-4267-af0c-121b4d57b8ba">

**Comment by vijayank88** [comment]
@maliberty can you check this?

**Comment by rovinski** [comment]
> I changed the contents of PDK file

You're going to need to be much more specific. This error simply means that you created a cell in the LEF, and there is no GDS/OAS cell with the same name so it cannot be converted.

It's interesting to note that it's on via cells, but it requires more information about the changes to find out the root issue.

**Reply by 706697916** [reply]
I replaced all the code of .lib  .macro.lef  .tech.lef  .gds  .drf  .layermap as another commercial  PDK, but I keep the name of these files unchanged. And the config.mk file is also changed according to the new PDK files.  You said the via cell is created in the LEF, I wonder what is the meaning of "there is no GDS/OAS cell with the same name so it cannot be converted"? Where the generated via GDS/OAS cells  need to match?

**Reply by 706697916** [reply]
This error occurs when the merged file 6_1_merged.gds  6_final.def and the lyt file want to generate the 6_final.gds file. Maybe my klayout lyt file has some problem. Below is part of the .lyt file.
<img width="1867" alt="1f84bb661e554e2c2ba46b3a454497c" src="https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/assets/70256434/3e4a1e7d-a57c-4c60-9863-b124d519b32d">
<img width="1553" alt="8948a7fcfad7b8ebfe2fb1d50e28c3a" src="https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/assets/70256434/9b914e0c-28e2-4c8a-8ddb-7d84ed4f1719">

**Reply by rovinski** [reply]
Does your `6_final.def` file have a `VIAS` section? Something that looks like this
```
VIAS 3 ;
    - via1_2_960_340_1_3_300_300 + VIARULE Via1Array-0 + CUTSIZE 140 140  + LAYERS metal1 via1 metal2  + CUTSPACING 160 160  + ENCLOSURE 70 100 70 70  + ROWCOL 1 3  ;
    - via2_3_960_340_1_3_320_320 + VIARULE Via2Array-0 + CUTSIZE 140 140  + LAYERS metal2 via2 metal3  + CUTSPACING 180 180  + ENCLOSURE 70 70 70 70  + ROWCOL 1 3  ;
    - via3_4_960_340_1_3_320_320 + VIARULE Via3Array-0 + CUTSIZE 140 140  + LAYERS metal3 via3 metal4  + CUTSPACING 180 180  + ENCLOSURE 70 70 90 70  + ROWCOL 1 3  ;
END VIAS
```

And do the vias you see the error for appear in that section?

**Reply by 706697916** [reply]
sorry, what I said just now is not accurate. 
There are 7 cells that reports ERROR of no matching GDS/OAS cell, of which 2 ERROR cells can be found in 6_final.def file, and other 5 ERROR cells can be found in the tech.lef file.

**Reply by 706697916** [reply]
below is the def file that contains VIAS section and the lef file that contains other VIAS.
<img width="1539" alt="94753c04926fda3e6e46d4620741b19" src="https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/assets/70256434/73a4549a-ba8b-472f-bb35-1b639b94834b">
<img width="727" alt="333143336800b7d90e051bf1bbf486c" src="https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/assets/70256434/f191b7d7-12df-454c-a5a6-bdc215c8c56e">

**Reply by 706697916** [reply]
All the via cells in 6_final.def shows ERROR of no matching GDS/OAS cell

**Reply by 706697916** [reply]
Can anybody give a solution about this no matching ERROR?

**Reply by vijayank88** [reply]
@706697916 Are you using latest commit of ORFS?
Which version klayout used?

**Reply by 706697916** [reply]
I download the ORFS about six months ago, and the version of klayout is  0.27.10

**Reply by vijayank88** [reply]
I hope this is the cause to the error. Please update ORFS to latest and try again. Is docker based or local installation?
Because currently ORFS used klayout `0.28.8`.

**Reply by 706697916** [reply]
I forget how I install it.  I update the klayout to 0.28.8 and the ERROR still occurs, but when I use the command "make" again without clean files, it will give me a "6_final.gds" file, which seems not right, below is the 6_final.gds . I will update the QRFS by local installation and try again.
<img width="807" alt="1691066118618" src="https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/assets/70256434/380c54dd-d94c-428e-9183-4ed2ebf489ae">

**Reply by 706697916** [reply]
I reinstall the ORFS. But the same ERROR of no matching GDS/OAS cell occurs. If I use the command "make gui_final" after command "make", it shows me the gds file, that is strange.
<img width="1280" alt="1691115377069" src="https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/assets/70256434/57433626-8b83-4375-8f65-28c5fab493d1">

**Reply by vijayank88** [reply]
As you mentioned updated PDK files. So we need reproducible test case. Please attach a test case, to debug the same.

**Reply by 706697916** [reply]
It is a commercial PDK. I am not able to provide it. Maybe I have to develop it by myself.

**Reply by 706697916** [reply]
By the way, is there a way to put the pins of GDS in the location where I define?

**Reply by maliberty** [reply]
The gds is an output - you would control the pin placement in OR.  You can use the https://github.com/The-OpenROAD-Project/OpenROAD/tree/master/src/ppl#place-individual-pin command

**Reply by 706697916** [reply]
Thank you very much!
